### PR TITLE
Upgrade view component.

### PR DESCRIPTION
**Pull request by justinlittman**
# Why was this change made? 🤔

# How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: app/components/works/show/doi_component.rb [hunk 1]
```diff
@@ -35,12 +35,12 @@ def doi_later
...
         # This is a version draft, the collection must have been changed to "yes"
         # after this item was deposited.
-        'DOI will become available once a new version is deposited.'
+        'DOI will become available once a new version is deposited.'.html_safe
       end
 
       def doi_opt_out
         'A DOI has not been assigned to this item. You may edit this item and select ' \
-          '"Yes" for the DOI option if you would like to receive a DOI.'
+        '"Yes" for the DOI option if you would like to receive a DOI.'.html_safe # rubocop:disable Rails/OutputSafety
```
Comment (ndushay) [inline app/components/works/show/doi_component.rb:43]: do you need both `.html_safe` and the rubocop disable?

File: app/components/works/detail_state_display_component.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def initialize(work_version:)
 
     def call
       value = I18n.t(work_version.state, scope: 'work.state')
-      return value unless work_version.wait_state?
+      return value.html_safe unless work_version.wait_state? # rubocop:disable Rails/OutputSafety
```
Comment (ndushay) [inline app/components/works/detail_state_display_component.rb:14]: do you need both `.html_safe` and the rubocop disable?

File: app/components/works/state_display_component.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@ def initialize(work_version:)
 
     def call
       value = I18n.t(work_version.state, scope: 'work.state')
-      return value unless work_version.wait_state?
+      return value.html_safe unless work_version.wait_state? # rubocop:disable Rails/OutputSafety
```
Comment (ndushay) [inline app/components/works/state_display_component.rb:14]: ditto

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -19,6 +19,9 @@
 require 'capybara-screenshot/rspec'
 require 'cocina/rspec/factories'
 require 'axe-rspec'
+require 'view_component/test_helpers'
+require 'view_component/system_test_helpers'
```
Comment (ndushay) [inline spec/rails_helper.rb:23]: nice!

**Review by ndushay — APPROVED** [review]
that's a LOT of rubocopping to go with this change ...

approve modulo consideration of question.

File: app/components/works/detail_state_display_component.rb (hunk 2 above)
Comment (justinlittman) [reply app/components/works/detail_state_display_component.rb:14]: Yes, the rubocop disable is saying "allow the use of html_safe".

**Comment by justinlittman** [comment]
Those actually were changes due to changes in the gem, not rubo.

**Pull request merged**